src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -1275,6 +1288,13 @@ private string LexIdentCore(out bool fDelimiterStart, out bool fDelimiterEnd)
                         fDelimiterEnd = false;
                         break;
                     }
+                    else if (_lex.TryGetPunctuator(CurrentChar.ToString(), out var tid) && tid == TokKind.Semicolon)
```

Comment:
this won't quite work, add a test in a `;;` separator locale and you'll see the issue.